src/configure/configure.ts

```diff
@@ -116,6 +116,69 @@ class Orchestrator {
         }
     }
 
+    private async getAzureResource() {
+        var azureResourceSelector = ResourceSelectorFactory.getAzureResourceSelector(this.inputs.pipelineConfiguration);
+        var resource = null;
+        try {
+            resource = await azureResourceSelector.getAzureResource(this.inputs);
+            return resource;
+        }
+        catch (err) {
+            if (err.message === Messages.setupAlreadyConfigured) {
+                this.continueOrchestration = false;
+                return;
+            }
+            else {
+                throw err;
+            }
+        }
+    }
+
+    private async selectTemplate(resource: GenericResource, inputs: WizardInputs) {
+        switch (resource.type) {
+            case TargetResourceType.AKS:
+                let templateParameterHelper = new TemplateParameterHelper();
+                let template = this.inputs.potentialTemplates[0] as LocalPipelineTemplate;
+                inputs.pipelineConfiguration.template = this.inputs.potentialTemplates[0];
+                await templateParameterHelper.setParameters(template.parameters, inputs);
+                break;
+
+            case TargetResourceType.WebApp:
+                var shortlistedTemplates = [];
+                shortlistedTemplates = this.inputs.potentialTemplates.filter((template) => template.targetKind === resource.kind);
```

Comment:
template.targetKind can be null or empty as well

Reply:
For a web app scenario, for remote templates, we have defined targetKinds as follows :
```
export function getTargetKind(template: PipelineTemplateMetadata) {
    var targetKind: TargetKind;
    if (template.attributes.deployTarget.toLowerCase().includes("webapp")) {
        if (template.attributes.deployTarget.toLowerCase().includes("container")) {
            targetKind = TargetKind.LinuxContainerApp;
        } else if (template.attributes.deployTarget.toLowerCase().includes("linux")) {
            targetKind = TargetKind.LinuxApp;
        } else if (template.attributes.deployTarget.toLowerCase().includes("windows")) {
            targetKind = TargetKind.WindowsApp;
        }
    }
    else {
        targetKind = null;
    }
    return targetKind;
}
```

And for local templates, only "Build and Test Python Django App" has targetKind as null. Not sure if it will be a part of webApp scenario.